### PR TITLE
CB-13684 Add error handling and additional logging to LoggingUsageRep…

### PR DESCRIPTION
**Pull request by nadamdb**
…orter

See detailed description in the commit message.

### CODE REVIEW

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java [hunk 1]
```diff
@@ -24,100 +24,160 @@
...
     }
 
     @Override
     public void cdpDatahubStatusChanged(UsageProto.CDPDatahubStatusChanged details) {
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatahubStatusChanged(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
```
Comment (akanto) [inline usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java:147]: wouldn't it be better to move this into the log method?

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java [hunk 2]
```diff
@@ -24,100 +24,160 @@
...
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpDatahubStatusChanged(details)
+                    .build());
+        } catch (Exception e) {
+            BINARY_EVENT_LOGGER.warn("Could not log binary format for the following usage event: {}! Cause: {}", details, e.getMessage());
```
Comment (akanto) [inline usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java:153]: can we log these with a different logger, not with the  BINARY_EVENT_LOGGER?

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java [hunk 3]
```diff
@@ -24,100 +24,160 @@
...
-        checkNotNull(details);
-        log(eventBuilder()
-                .setCdpDatahubStatusChanged(details)
-                .build());
+        try {
+            BINARY_EVENT_LOGGER.info("Binary format logging for the following usage event: {}", details);
+            checkNotNull(details);
+            log(eventBuilder()
+                    .setCdpDatahubStatusChanged(details)
+                    .build());
```
Comment (akanto) [inline usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java:151]: would be nice to have  logline before the catch

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java (hunk 1 above)
Comment (nadamdb) [reply usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java]: done

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java (hunk 2 above)
Comment (nadamdb) [reply usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java]: done

File: usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java (hunk 3 above)
Comment (nadamdb) [reply usage-collection/src/main/java/com/sequenceiq/cloudbreak/usage/LoggingUsageReporter.java:153]: done

**Review by akanto — APPROVED** [review]

**Pull request merged**